lib/mongoDb/MongoDBTransaction.js

```diff
@@ -0,0 +1,87 @@
+const InvalidSessionError = require('./errors/InvalidSessionError');
+
+const UNKNOWN_TRANSACTION_COMMIT_RESULT = 'UnknownTransactionCommitResult';
+const TRANSIENT_TRANSACTION_ERROR = 'TransientTransactionError';
+
+class MongoDBTransaction {
+  /**
+   *
+   * @param {MongoClient} mongoClient
+   */
+  constructor(mongoClient) {
+    this.mongoClient = mongoClient;
+    this.session = null;
+  }
+
+  /**
+   * Start new transaction
+   *
+   * @returns {Promise<void>}
+   */
+  async start() {
+    if (this.session) {
+      throw new InvalidSessionError('Session is already started');
```

Comment:
We can keep message inside the error class. I think we need two types of error: started and not started.